### PR TITLE
Remember ATB order

### PR DESCRIPTION
**Pull request by rueter37**
This PR remembers the ATB order now if two or more actors have a full ATB gauge.

Fixes: #2775

### CODE REVIEW

**Review by fdelapena — APPROVED** [review]

File: src/scene_battle_rpg2k3.cpp [hunk 1]
```diff
@@ -735,14 +735,27 @@ static bool BattlerReadyToAct(const Game_Battler* battler) {
...
 }
 
-static int GetFirstReadyActor() {
+void Scene_Battle_Rpg2k3::UpdateReadyActors() {
 	const auto& actors = Main_Data::game_party->GetActors();
 	for (size_t i = 0; i < actors.size(); ++i) {
 		auto* actor = actors[i];
 		if (BattlerReadyToAct(actor)) {
-			return i;
+			if (std::find(atb_order.begin(), atb_order.end(), i) == atb_order.end()) {
```
[the remark this answers is not in the document]
Comment (rueter37) [reply src/scene_battle_rpg2k3.cpp:743]: Any suggestions how to solve this? It's very likely that I will mess this up further if I try to fix this on my own.

[the remark this answers is not in the document]
Comment (rueter37) [reply src/scene_battle_rpg2k3.cpp]: This should work now. Thanks for the fix @Ghabry!

[the remark this answers is not in the document]
Comment (Ghabry) [reply src/scene_battle_rpg2k3.cpp]: oh, I missed that the Actor ID is not the same as the index in that array.

Though I'm still unsure if this atb_order can get stale when an event command adds or removes actors. This needs more careful testing but it 99% at the goal.

**Review by carstene1ns — APPROVED** [review]
Looks good to me.

**Pull request merged**